src/main/java/application/rest/v1/configmaps/KindActionMappingProcessor.java

```diff
@@ -0,0 +1,462 @@
+                                String kamNamespace = getKAMNamespace(item);
+                                if (element != null) {
+                                    JsonObject spec = element.getAsJsonObject();
+                                    int precedenceIndex = spec.get(PRECEDENCE_PROPERTY_NAME).getAsInt()-1;
+                                    JsonArray mappings = spec.getAsJsonArray(MAPPINGS_PROPERTY_NAME);
+
+                                    // iterate through each mapping within a KAM resource
+                                    mappings.forEach(mapItem-> {
+                                        if (mapItem != null) {
+                                            JsonObject props = mapItem.getAsJsonObject();
+                                            if (props != null) {
+                                                JsonElement prop = props.get(APIVERSION_PROPERTY_NAME);
+                                                String apiVersion = (prop != null) ? prop.getAsString():null;
+                                                prop = props.get(NAME_PROPERTY_NAME);
+                                                String name = (prop != null) ? prop.getAsString():null;
+                                                prop = props.get(SUBKIND_PROPERTY_NAME);
+                                                String subkind = (prop != null) ? prop.getAsString():null;
+                                                prop = props.get(KIND_PROPERTY_NAME);
+                                                String kind = (prop != null) ? prop.getAsString():null;
+                                                prop = props.get(MAPNAME_PROPERTY_NAME);
+                                                String mapname = (prop != null) ? prop.getAsString():null;
+                                                Logger.log(className, "getConfigMapsFromKAMs", Logger.LogType.DEBUG,
+                                                           "\nmapping info: " + 
+                                                           "\napiVersion = " + apiVersion +
+                                                           "\nname = " + name +
+                                                           "\nsubkind = " + subkind +
+                                                           "\nkind = " + kind + 
+                                                           "\nmapname = " + mapname);
+
+                                                if (isApiVersionMatch(apiVersion, compApiVersion)) {
+                                                    int compPropsIdx = exemineMappingProperties(compName, compSubkind, compKind);
+                                                    int kamMappingPropIdx = exemineMappingProperties(name, subkind, kind);
+                                                    boolean found = foundMatchedConfigMap(compPropsIdx, kamMappingPropIdx, 
```

Comment:
Maybe use a clearer method name. "Found" is past tense:  the configmap is not found yet.  In fact, the method does not examine configmaps at all.  What the method is trying to do is determine if the subject resource maps to the current KAM mapping rule.   A method name that more specifically identifies the task might be:   

```
boolean matches= resourceMatchesRule(....)
```